lib/jsonapi_compliable/adapters/active_record/base.rb

```diff
@@ -2,9 +2,124 @@ module JsonapiCompliable
   module Adapters
     module ActiveRecord
       class Base < ::JsonapiCompliable::Adapters::Abstract
-        # (see Adapters::Abstract#filter)
-        def filter(scope, attribute, value)
-          scope.where(attribute => value)
+        def filter_string_eq(scope, attribute, value, is_not: false)
+          column = scope.klass.arel_table[attribute]
+          clause = column.lower.eq_any(value.map(&:downcase))
+          is_not ? scope.where.not(clause) : scope.where(clause)
+        end
+
+        def filter_string_eql(scope, attribute, value, is_not: false)
+          clause = { attribute => value }
+          is_not ? scope.where.not(clause) : scope.where(clause)
+        end
+
+        def filter_string_not_eq(scope, attribute, value)
+          filter_string_eq(scope, attribute, value, is_not: true)
+        end
+
+        def filter_string_not_eql(scope, attribute, value)
+          filter_string_eql(scope, attribute, value, is_not: true)
+        end
+
+        def filter_string_prefix(scope, attribute, value, is_not: false)
+          column = scope.klass.arel_table[attribute]
+          map = value.map { |v| "#{v}%" }
+          clause = column.lower.matches_any(map)
+          is_not ? scope.where.not(clause) : scope.where(clause)
+        end
+
+        def filter_string_not_prefix(scope, attribute, value)
+          filter_string_prefix(scope, attribute, value, is_not: true)
+        end
+
+        def filter_string_suffix(scope, attribute, value, is_not: false)
+          column = scope.klass.arel_table[attribute]
+          map = value.map { |v| "%#{v}" }
+          clause = column.lower.matches_any(map)
+          is_not ? scope.where.not(clause) : scope.where(clause)
+        end
+
+        def filter_string_not_suffix(scope, attribute, value)
+          filter_string_suffix(scope, attribute, value, is_not: true)
+        end
+
+        def filter_string_like(scope, attribute, value, is_not: false)
+          column = scope.klass.arel_table[attribute]
+          map = value.map { |v| "%#{v.downcase}%" }
+          clause = column.lower.matches_any(map)
+          is_not ? scope.where.not(clause) : scope.where(clause)
+        end
+
+        def filter_string_not_like(scope, attribute, value)
+          filter_string_like(scope, attribute, value, is_not: true)
+        end
+
+        def filter_integer_eq(scope, attribute, value, is_not: false)
+          clause = { attribute => value }
+          is_not ? scope.where.not(clause) : scope.where(clause)
+        end
+        alias :filter_float_eq :filter_integer_eq
```

Comment:
You’re defining these as adapter methods but I think there might be a set of missing classes. If you had, say JsonapiCompliable::Filters::ActiveRecord::Number, ::String, etc, then it becomes easier to subclass instead of aliasing and I think supporting new types in the future might be a bit easier.

Reply:
I'll take a stab when I circle back around...it's a good idea, but I think probably more an implementation detail where these methods still exist. One common use case is to implement an incomplete adapter (like subclassing the NullAdapter but supporting string equals and thats it). In that case the class style would be pretty heavyweight ceremony IMO. I think it's a good idea to use classes underneath the hood of these methods, or maybe check to see if a class exists before falling back to the method name, though. Will go down this route in an upcoming refactor pass and see how well it works out.